### PR TITLE
Override Gemfile.lock while running tests

### PR DESCRIPTION
**Pull request by naps62**
Why:
- When running the tests, a new app is generated, and `bundle install`
  needs to resolve dependencies from scratch.
  If a predefined Gemfile.lock exists, `bundler` will only need to check
  if the specified versions exist (and install the ones that don't),
  instead of taking forever to ask rubygems for this info.

### CODE REVIEW

**Comment by naps62** [comment]
@zamith can you replace the Gemfile.lock in this PR with one generated from your dummy app?

Mine doesn't include capybara-webkit, which makes the dummy specs fail

File: lib/firestarter/app_builder.rb [hunk 1]
```diff
@@ -308,13 +308,19 @@ def setup_metrics
       copy_file "metrics.reek", ".metrics.reek"
     end
 
+    def override_gemfile_lock_for_tests
+      return unless ENV["TESTING"]
```
Comment (zamith) [inline lib/firestarter/app_builder.rb]: I'm not a big fan of this, because we are writing "production" code only for the purpose of testing. The precedent was set by `override_path_for_tests`, but I just removed it in this branch, because there is no need for it to be in the `app_builder`.

Also, I'm concerned that because we have a `Gemfile.lock` we won't be testing in the same exact scenario it will be run, which kind of defeats the point of testing.

I understand that it is a bit painful to have to wait so long for tests, but I don't think this is the way to speed that up and still have a reliable test suite. We might be able to get around this concern by only using the `Gemfile.lock` locally, and not in CI, but I don't see how to get around the fact that this is code just for the tests.

Comment (pfac) [inline lib/firestarter/app_builder.rb]: I suggest splitting the issue here. We have the precedent set by `override_path_for_tests` which seems to be wrong. Fixing it in this PR is kinda out of scope, so I would suggest moving it to another PR. What do you think @zamith?

**Comment by pfac** [comment]
@zamith has a point there regarding the fact that we are using a Gemfile we won't be testing in the scenario it is supposed to be used. Also, this is not a problem when the tests are running in the CI, so it is not something we would want in all test scenarios.

**Review by pfac — CHANGES_REQUESTED** [review]
@zamith has a point there regarding the fact that we are using a Gemfile we won't be testing in the scenario it is supposed to be used. Also, this is not a problem when the tests are running in the CI, so it is not something we would want in all test scenarios.
